service/core/kafka/server/src/main/scala/com/lightbend/lagom/internal/broker/kafka/Producer.scala

```diff
@@ -101,6 +101,8 @@ private[lagom] object Producer {
     def generalHandler: Receive = {
       case Failure(e) =>
         throw e
+      case t: Throwable =>
+        throw t
```

Comment:
Have you tested this?

From the Akka docs on pipe:

> The successful result of the future is sent as a message to the recipient, or the failure is sent in a `akka.actor.Status.Failure` to the recipient.

So, that's what the case above this is supposed to be matching. That said, it may be matching a `scala.util.Failure`, which is a different failure, and so wouldn't match.